elastic/beats
# [Filebeat] Get AWS credentials again when token expired

## What does this PR do?

This PR is to add `GetAWSCredentials` function to get credentials again when previous AWS credentials are expired or invalid.

## Why is it important?

This will avoid users have to manually stop and restart Filebeat in order to obtain the new AWS credentials.

## Checklist


## How to test this PR locally

Use `~/.aws/credentials` to store AWS credentials and then in Filebeat, change filebeat.yml to use s3 input:
```
filebeat.inputs:
- type: s3
  queue_url:   https://sqs.us-east-1.amazonaws.com/428152502467/test-fb-ks
  credential_profile_name: elastic-beats
```

Before starting Filebeat, when generating temporary credentials, set `DurationSeconds` to be 900 seconds. Wait after 900 seconds(15min) and then start filebeat with `./filebeat -e`. You should see error message immediately:
```
2020-03-24T11:51:37.872-0600    WARN    [s3]    s3/input.go:206 credentials are invalid, acquiring AWS credentials again from config: ExpiredToken: The security token included in the request is expired
        status code: 403, request id: 702080a4-8d34-5d5a-b212-d650c7ddfa53
```
Then generate a new set of temporary credentials and update credentials in the file `~/.aws/credentials`. This error message should disappear right away.

## Related issues

- Closes https://github.com/elastic/beats/issues/17189
